### PR TITLE
exercises 3.1 and 3.2

### PR DESCRIPTION
**Pull request by MayaStelzer**

### CODE REVIEW

File: work/Maya/exercise3_1.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        'B': [5, 6, 7, 8],
+        'C': [9, 10, 11, 12]}
+df = pd.DataFrame(data)
+
+
+def calculate_mean(df, result, lock, column):
+    start_time = time()
+    column_avg = df[column].mean()
+    end_time = time()
+    lock.acquire()
```
Comment (Alvaro8gb) [inline work/Maya/exercise3_1.py]: lock dont needed

File: work/Maya/exercise3_1.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import pandas as pd
+import threading
+from time import time
+
+data = {'A': [1, 2, 3, 4],
+        'B': [5, 6, 7, 8],
```
Comment (Alvaro8gb) [inline work/Maya/exercise3_1.py:6]: the code I provide have a dataset more big, but okey

File: work/Maya/exercise3_2.py [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import subprocess
+
+command = ['nice', '-19', 'sleep', '30']
```
Comment (Alvaro8gb) [inline work/Maya/exercise3_2.py:3]: nice

File: work/Maya/exercise4_1.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    if not os.path.exists(file_path):
+        print(f"{file_path} does not exist")
+        return
+    permission = os.stat(file_path).st_mode
+    print(f"permission of {file_path} is {permission}")
+    print(f"Write permission: {'Yes' if permission & 0o200 else 'No'}")
+
+if __name__ == "__main__":
+    if len(sys.argv) != 2:
+        print("Need the file path as an argument")
```
Comment (Alvaro8gb) [inline work/Maya/exercise4_1.py]: **Note:** Consider allowing the user to input the file path interactively if no argument is provided.

File: work/Maya/exercise4_1.py [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import os
+import sys
+
+def display_permission(file_path):
+    if not os.path.exists(file_path):
+        print(f"{file_path} does not exist")
+        return
+    permission = os.stat(file_path).st_mode
+    print(f"permission of {file_path} is {permission}")
```
Comment (Alvaro8gb) [inline work/Maya/exercise4_1.py:9]: good

**Review by Alvaro8gb — CHANGES_REQUESTED** [review]
It good, but can be better

File: work/Maya/exercise3_1.py (hunk 2 above)
Comment (MayaStelzer) [reply work/Maya/exercise3_1.py:6]: I couldn't find the dataset you provided. Is it on teams or github?

**Pull request closed without merging**